### PR TITLE
Migrate to postgres - refactor models and tests

### PR DESCRIPTION
**Pull request by SammyAbukmeil**
- Add `pg` package and remove `mysql2`
- Different syntax to get rows
  - Update models to 
  - Update tests
---

**Note**: Following this work, we should look for any opportunities to reduce the amount of queries via postgres' `RETURNING id` (and if this exists: `RETURNING *`)
- Let's create an issue for this once this is merged in

---

**Note**: You'll need to install Postgres on your machine, I used the official installer via the website (you could also use homebrew).
- During installation, it will ask if you want a tool called "Stack Builder" - from what I read online this is optional for installing extensions; I decided not to install it.

### CODE REVIEW

File: server/db/connection.js [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
   user: process.env.DB_TEST_USER,
   password: process.env.DB_TEST_PASSWORD,
   host: process.env.DB_TEST_HOST,
+  port: process.env.DB_TEST_PORT,
 };
 
 const config = process.env.NODE_ENV === "test" ? testDatabase : database;
 
-export default await mysql.createConnection(config);
+const clinet = new Client(config);
```
Comment (michaelti) [inline server/db/connection.js]: Typo: should say `client`

File: server/db/connection.js [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
   host: process.env.DB_HOST,
+  port: process.env.DB_PORT,
 };
 
 const testDatabase = {
   database: process.env.DB_TEST_DATABASE,
   user: process.env.DB_TEST_USER,
   password: process.env.DB_TEST_PASSWORD,
   host: process.env.DB_TEST_HOST,
+  port: process.env.DB_TEST_PORT,
```
Comment (michaelti) [inline server/db/connection.js:18]: This should be added to `.env.example`

File: server/routes/matchRoutes.test.js [hunk 3]
```diff
@@ -202,18 +202,17 @@ describe("POST /matches", () => {
     await cleanupMatches();
     await request(app).post("/matches").send(body);
 
-    const [[matchesPlayersRecord]] = await db.execute(
-      "SELECT points_given FROM matches_players WHERE player_id = ?",
+    const matchesPlayers = await db.query(
```
Comment (michaelti) [inline server/routes/matchRoutes.test.js:205]: Thank you for replacing the abomination of `const [[matchesPlayersRecord]]`

**Review by michaelti — CHANGES_REQUESTED** [review]
For some reason, the POST endpoint tests failed when I first ran `npm run test`
- They were responding `500` instead of `201`.
- I then tried `npm run test-watch`, and that worked.
- From that point, `npm run test` has been passing and I couldn't reproduce the issue...

I say let's circle back to that later... let's SHIP IT 🚀

PS. I've installed Postgres using DBngin. From the creators of TablePlus.

File: server/db/connection.js (hunk 2 above)
Comment (SammyAbukmeil) [reply server/db/connection.js:18]: Done https://github.com/michaelti/play-up/pull/36/commits/270850f0c9923ae08fed2045ba43d6e30f6124a3

File: server/db/connection.js (hunk 1 above)
Comment (SammyAbukmeil) [reply server/db/connection.js]: Done https://github.com/michaelti/play-up/pull/36/commits/270850f0c9923ae08fed2045ba43d6e30f6124a3

File: server/routes/matchRoutes.test.js (hunk 3 above)
Comment (SammyAbukmeil) [reply server/routes/matchRoutes.test.js:205]: :)

**Review by michaelti — APPROVED** [review]

**Pull request merged**